src/js/stores/AppsStore.js

```diff
@@ -80,6 +81,41 @@ function processApps(apps) {
   }).value();
 }
 
+function applyAppDelayStatus(app, queue) {
+  var changed = false;
+  var status;
+
+  var queueEntry = lazy(queue).find(function (entry) {
+    return entry.app != null && app.id === entry.app.id && entry.delay != null;
+  });
+
+  if (queueEntry) {
+    if (queueEntry.delay.overdue === false
+        && queueEntry.delay.timeLeftSeconds > 0) {
+      status = AppStatus.DELAYED;
+    } else if (queueEntry.delay.overdue === true) {
+      status = AppStatus.WAITING;
+    }
+
+    if (status) {
+      if (app.status !== status) {
+        changed = true;
+      }
+      app.status = status;
+    }
+  }
+
+  return changed;
```

Comment:
I don't see the point of that. `changed` has a default assignment of `false` and is only assigned boolean values.
Maybe we could rename the variable to `hasChanged` to make it explicit that this value should only hold boolean values?